### PR TITLE
Small refactorings

### LINKED ISSUE #1759: {pathinfo: true} can break JS syntax in built files when the loader query contains */
Here's a piece of JS that was emitted by webpack:

	/* WEBPACK VAR INJECTION */(function(global) {module.exports = global["MobileRouter"] = __webpack_require__(/*! -!/Users/mikolajdadela/code/lb-react-frontend/~/babel-loader?{"plugins":["transform-react-jsx"],"ignore":["node_modules/**/*"]}!./mobilePages/MobileRouter.jsx */ 1);
	/* WEBPACK VAR INJECTION */}.call(exports, (function() { return this; }())))

A shortened version of the first line:

	/* WEBPACK ... */( ... __webpack_require__(/* ... ["node_modules/**/*"] ...  */ 1);
                                                                       ^

When I run this code in browser, I get

    Uncaught SyntaxError: Unexpected token *

### PR DESCRIPTION
**Pull request by sokra**
**What kind of change does this PR introduce?**
refactor

**Did you add tests for your changes?**
existing tests

**If relevant, link to documentation update:**
N/A

**Summary**
helper functions in Template (fixes #1759)
use getReference instead of `dep.module` + `dep.weak`

**Does this PR introduce a breaking change?**
no

**Other information**

### CODE REVIEW

File: lib/AmdMainTemplatePlugin.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class AmdMainTemplatePlugin {
 				typeof m.request === "object" ? m.request.amd : m.request
 			));
 			const externalsArguments = externals.map((m) =>
-				Template.toIdentifier(`__WEBPACK_EXTERNAL_MODULE_${m.id}__`)
+				`__WEBPACK_EXTERNAL_MODULE_${Template.toIdentifier(`${m.id}`)}__`
```
Comment (TrySound) [inline lib/AmdMainTemplatePlugin.js]: Why template literal around `m.id`?

File: lib/ExternalModule.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ class ExternalModule extends Module {
 	}
 
 	getSourceForAmdOrUmdExternal(id, optional, request) {
-		const externalVariable = Template.toIdentifier(`__WEBPACK_EXTERNAL_MODULE_${id}__`);
+		const externalVariable = `__WEBPACK_EXTERNAL_MODULE_${Template.toIdentifier(`${id}`)}__`;
```
Comment (TrySound) [inline lib/ExternalModule.js]: And here dumb template literal

File: lib/dependencies/ImportDependency.js [hunk 3]
```diff
@@ -22,20 +23,12 @@ ImportDependency.Template = class ImportDependencyTemplate {
...
 		const depBlock = dep.block;
 		const promise = DepBlockHelpers.getDepBlockPromise(depBlock, outputOptions, requestShortener, "import()");
-		const comment = this.getOptionalComment(outputOptions.pathinfo, requestShortener.shorten(dep.request));
+		const comment = outputOptions.pathinfo ? Template.toComment(requestShortener.shorten(dep.request)) : "";
 
 		const content = this.getContent(promise, dep, comment);
 		source.replace(depBlock.range[0], depBlock.range[1] - 1, content);
 	}
 
-	getOptionalComment(pathinfo, shortenedRequest) {
```
Comment (niieani) [inline lib/dependencies/ImportDependency.js]: @sokra note that this is a breaking change for plugins using this api (`getOptionalComment`)

File: lib/AmdMainTemplatePlugin.js (hunk 1 above)
Comment (sokra) [inline lib/AmdMainTemplatePlugin.js]: To convert it to a string. id can be a number

File: lib/dependencies/ImportDependency.js (hunk 3 above)
Comment (sokra) [inline lib/dependencies/ImportDependency.js]: I marked it as breaking change.

File: lib/AmdMainTemplatePlugin.js (hunk 1 above)
Comment (sokra) [inline lib/AmdMainTemplatePlugin.js]: An alternative would be `m.id + ""` but that looks ugliy too...

Comment (probablyup) [inline lib/AmdMainTemplatePlugin.js]: `String(m.id)` works too and is clear

**Pull request merged**